SConstruct

```diff
@@ -1015,7 +1015,7 @@ if env["compiledb"] and env.scons_version < (4, 0, 0):
     # Generating the compilation DB (`compile_commands.json`) requires SCons 4.0.0 or later.
     print_error("The `compiledb=yes` option requires SCons 4.0 or later, but your version is %s." % scons_raw_version)
     Exit(255)
-if env.scons_version >= (4, 0, 0):
+if env["compiledb"] and env.scons_version >= (4, 0, 0):
```

Comment:
I'd put this and the previous check in the same `if env["compiledb"]`. I'm pretty this was the case originally but ut must have been broken in a recent refactor.

Reply:
Sounds good. I merged the 2 blocks.